src/Wellcome.Dds/Wellcome.Dds.AssetDomainRepositories/Ingest/CloudServicesIngestRegistry.cs

```diff
@@ -0,0 +1,179 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using Wellcome.Dds.AssetDomain.Dlcs.Ingest;
+using Wellcome.Dds.AssetDomain.Mets;
+
+namespace Wellcome.Dds.AssetDomainRepositories.Ingest
+{
+
+    public class CloudServicesIngestRegistry : IIngestJobRegistry
+    {
+        private readonly IMetsRepository _metsRepository;
+        private readonly DdsInstrumentationContext ddsInstrumentationContext;
+
+        public CloudServicesIngestRegistry(
+            IMetsRepository metsRepository,
+            DdsInstrumentationContext ddsInstrumentationContext)
+        {
+            _metsRepository = metsRepository;
+            this.ddsInstrumentationContext = ddsInstrumentationContext;
+        }
+
+        public IEnumerable<DlcsIngestJob> GetJobs(int number)
+        {
+            return ddsInstrumentationContext.DlcsIngestJobs
+                .OrderByDescending(j => j.Created)
+                .Take(number)
+                .Include(j => j.DlcsBatches)
+                .ToArray(); // The operation cannot be completed because the DbContext has been disposed. Want to defer this
+        }
+
+        public DlcsIngestJob GetJob(int id)
+        {
+            return ddsInstrumentationContext.DlcsIngestJobs
+                .Include(j => j.DlcsBatches)
+                .FirstOrDefault(j => j.Id == id);
+        }
+
+        public IEnumerable<DlcsIngestJob> GetQueue(DateTime? after)
+        {
+            var query = ddsInstrumentationContext.DlcsIngestJobs
+                .OrderByDescending(j => j.Created)
+                .Include(j => j.DlcsBatches)
+                .Where(j => j.StartProcessed == null);
+
+            if (after.HasValue)
+            {
+                return query.Where(j => j.Created > after.Value).ToArray();
+            }
+            return query.ToArray();
+        }
+
+        public IEnumerable<DlcsIngestJob> GetProblems(DateTime? after)
```

Comment:
Can `after` go? Not used here and commented out in the original version.

Reply:
There is a use in another utility:
https://github.com/wellcomelibrary/dds-ecosystem/blob/new-storage-service/wellcome-dds/DlcsBackgroundReconciler/Program.cs#L164

However, I think we should still drop this and reintroduce if we need to go back.

Reply:
However... I will turn the Take(..) arg into a parameter.